crates/ef-testing-proc-macro/src/content_reader.rs

```diff
@@ -0,0 +1,54 @@
+use std::collections::BTreeMap;
+
+use serde_json::Value;
+
+use crate::{dir_reader::PathWrapper, utils::blockchain_tests_to_general_state_tests_path};
+
+pub struct ContentReader;
+
+impl ContentReader {
+    pub fn secret_key(path: PathWrapper) -> Result<Value, eyre::Error> {
+        let path = blockchain_tests_to_general_state_tests_path(path);
+        let content = path.read_file_to_string()?;
+
+        let cases: BTreeMap<String, Value> = serde_json::from_str(&content)?;
+        let case = cases
+            .into_values()
+            .collect::<Vec<_>>()
+            .first()
+            .ok_or_else(|| eyre::eyre!("No case found"))?
+            .clone();
+
+        Ok(case
+            .get("transaction")
+            .ok_or_else(|| eyre::eyre!("No transaction found"))?
+            .get("secretKey")
+            .ok_or_else(|| eyre::eyre!("No secret key found"))?
+            .clone())
+    }
+
+    pub fn pre_state(test_case: &Value) -> Result<Value, eyre::Error> {
+        Ok(serde_json::from_value(
+            test_case
+                .get("pre")
+                .ok_or_else(|| eyre::eyre!("No pre state found"))?
+                .clone(),
+        )?)
+    }
+
+    pub fn post_state(test_case: &Value) -> Result<Value, eyre::Error> {
```

Comment:
Thinking out loud, I'm okay with eyre for this PR, but it's preferable to use custom errors with thiserror and use eyre only in the highest function, such as main.rs, it's quite compact to write an error enum imo

Reply:
I would also normally use thiserror, but since all the functions here are only used to generate the code for the macro, it felt wasteful to have a real nice error enum when there is little error handling going on.

Reply:
Ah I see, how do other people do?